x-pack/legacy/plugins/siem/package.json

```diff
@@ -11,10 +11,10 @@
   },
   "devDependencies": {
     "@types/lodash": "^4.14.110",
-    "@types/js-yaml": "^3.12.1",
     "@types/react-beautiful-dnd": "^11.0.3"
   },
   "dependencies": {
+    "js-yaml": "^3.13.1",
```

Comment:
do we need that at all? `js-yaml` is installed in main package.json https://github.com/elastic/kibana/blob/master/package.json#L187

Reply:
I thought exactly the same as you, but when I tried to commit the changes I had a linter error `import/no-extraneous-dependencies` this is the only way I found to solve it, if you have any idea about how to fix that linter error, please let me know :)

Reply:
@FrankHassanabad are you aware of a better way?

Reply:
I just checked this pr out @MadameSheema and:
* removed these deps
* Ran a kbn clean
* Ran a bootstrap
* Modified the `plugins/siem/cypress/support/commands.ts`
* Did a commit locally

And everything works! So 🎉 !!

The reason why you shouldn't need this and it is going to work out for you is if you look at your awesome changes from JavaScript to TypeScript of this file: 

`/plugins/siem/cypress/support/commands.js` to become
`/plugins/siem/cypress/support/commands.ts`

I am guessing at some point you might have had this as actually a require when that file was a js file?

```
import yaml from 'js-yaml';
```

When they're js files you have to add a:

```ts
// eslint-disable-next-line import/no-extraneous-dependencies
```

For example, `x-pack/legacy/plugins/siem/cypress/plugins/index.js`

<img width="1096" alt="Screen Shot 2019-12-03 at 8 59 50 AM" src="https://user-images.githubusercontent.com/1151048/70067350-5c477800-15ab-11ea-9203-c8b87268b7c6.png">

That's how you get the issue _only_ when they're JS files. And that should be what you use with `js` files or anything else linter related is to turn it off if the rule its self is wrong.

But great news! Since you moved that file over to `ts` it looks like you do not need to mess with the `package.json` file. You can safely remove this and not worry about modifying `ts` files and getting linter errors on a commit. If you do (for TypeScript files), let me know.

If you're using require in `js` files, then  just turn off the linter error through:

```ts
// eslint-disable-next-line import/no-extraneous-dependencies
```

and then do the commit.

Reply:
I knew he'd know 🎉